scapy/layers/dns.py

```diff
@@ -369,6 +369,9 @@ def i2len(self, pkt, x):
     def i2m(self, pkt, s):
         ret_s = b""
         for text in s:
+            if not text:
+                ret_s += b"\x00"
```

Comment:
I kinda dislike this. Empty strings have a `\x00` appended, but not regular strings. This is weird and to me inconsistent

Reply:
Empty strings are common in the mDNS world where services can be advertised without any key/value pairs. According to RFC6763
>    An empty TXT record containing zero strings is not allowed [RFC1035].DNS-SD implementations MUST NOT emit empty TXT records.

Originally it came up in https://github.com/lathiat/avahi/pull/490 where I used scapy to edit mDNS packets sent by avahi and systemd-resolved. Without this patch those packets lose empty strings.

Reply:
> Empty strings have a \x00 appended, but not regular strings

`\x00` isn't appended to empty strings. It shows that their length is zero and they are followed by zero characters. It's consistent with non-empty strings like `\x03abc` (where 3 is its length and it's followed by the-three byte string).

Reply:
I updated the commit message to clarify that `\x00` shows that the length of empty strings is zero and it isn't appended to actual strings.

Reply:
FWIW it can be cross-checked with Wireshark by running something like
```python
tdecode(Ether()/IP()/UDP()/DNS(an=[DNSRR(rrname='H.local', type='TXT', rdata=["", "", "", ""])]))
```
Without this patch all the empty strings get lost in the sense that the TXT RR in the answer section is empty:
```sh
    Answers
        H.local: type TXT, class IN
            Name: H.local
            Type: TXT (Text strings) (16)
            Class: IN (0x0001)
            Time to live: 0 (0 seconds)
            Data length: 0
```
With this patch applied all the four empty string are included in the TXT RR:
```sh
    Answers
        H.local: type TXT, class IN
            Name: H.local
            Type: TXT (Text strings) (16)
            Class: IN (0x0001)
            Time to live: 0 (0 seconds)
            Data length: 4
            TXT Length: 0
            TXT: 
            TXT Length: 0
            TXT: 
            TXT Length: 0
            TXT: 
            TXT Length: 0
            TXT: 
```

Reply:
Thanks for clarifying. I understood this PR completely wrong, it makes a ton of sense.